Code/EnginePlugins/JoltPlugin/System/JoltCore.cpp

```diff
@@ -77,34 +79,59 @@ void ezJoltCore::DebugDraw(ezWorld* pWorld)
 #endif
 }
 
+ezMutex s_AllocSizeMutex;
+
 void* ezJoltCore::JoltMalloc(size_t inSize)
 {
-  return ezJoltCore::s_pAllocator->Allocate(inSize, 16);
+  void* ptr = ezJoltCore::s_pAllocator->Allocate(inSize, 8);
+
+  EZ_LOCK(s_AllocSizeMutex);
+  s_AllocSizes[ptr] = inSize;
+  return ptr;
 }
 
 void ezJoltCore::JoltFree(void* inBlock)
 {
+  EZ_ASSERT_DEBUG(inBlock != nullptr, "");
```

Comment:
Why are the sizes not removed here? This means the hashtable is getting bigger and bigger over lifetime.